app/ont/ont.rb

```diff
@@ -24,9 +30,18 @@ def self.direct_request_attributes
     request_attributes - associated_request_attributes
   end
 
-  # Parameters would be request: and library_attributes: if this was supported.
-  def self.library_factory(*)
-    raise StandardError, 'Unsupported' # Only Pacbio is supported at the moment
+  def self.library_factory(request:, library_attributes:)
+    # Get the tag_set from the kit_barcode
+    ont_tag_set = TagSet.find_by(name: library_attributes[:kit_barcode])
+    # Find the tag_id from the tag_set based on the tag_sequence/oligo
+    library_attributes[:tag_id] =
+      ont_tag_set&.tags&.find_by(oligo: library_attributes[:tag_sequence])&.id
+    filtered_attributes = library_attributes.slice(*self.library_attributes)
```

Comment:
In theory tag id can be nil if they are importing just one library in a pool so its a bit of a tricky situation. If there are multiple libraries and some dont have tags it will get caught later on in validation and error.